### ISSUE TITLE
Timezone plugin doesn't consider the timezone from the date param

### ISSUE THREAD

**Issue by superKalo** [iamkun/dayjs#1556]
**Describe the bug**
The timezone plugin doesn't take into consideration the timezone from the param.

**Expected behavior**

I'm in the middle of a journey to migrate from MomentJS to DayJS.

Here's MomentJS:

```js
moment.tz('2021-06-28T00:00:00+02:00', 'Europe/Berlin').format(); // "2021-06-28T00:00:00+02:00"
```

Same thing with DayJS:

```js
dayjs.tz('2021-06-28T00:00:00+02:00', 'Europe/Berlin').format(); // "2021-06-27T22:00:00+02:00"
// but in general - it varies based on local time zone (mine is GMT+03:00).
```

This is confusing in my opinion, since I'd expect DayJS to consider the timezone provided in the param, where there is one (just like MomentJS does).

**Information**
 - Day.js Version 1.10.5
 - OS: iOS and Android
 - Browser: React Native (Expo)
 - Time zone: GMT+03:00 is my current one

**Comment by philipRatsovPP** [comment]
Yeah same problem here:
I am using similar date as OP's example in unit tests and it results in a previous date. Tried to preserve the time zone which is passed in with something similar like moment.parseZone() (MomentJS) but there isn't any alternative in Dayjs. 
So I also agree that in general the timezone plugin is not working correctly when an time offset is passed in.

**Comment by addisonElliott** [comment]
This is a common misconception with Day.js and moment. The date is always converted to the machine's local timezone.

Moment has a `parseZone()` function that does what you want. Here's a related issue for Day.js: https://github.com/iamkun/dayjs/issues/651

**Comment by 7aklhz** [comment]
I'm facing a similar problem, and I'm not sure what you mean by "The date is always converted to the machine's local timezone" @addisonElliott . 

Here is my issue : I have a date, for example "07/22/2021" and I want to "tell" my machine that this date was created in a "New York timezone". 

Do do so, I'm doing "dayjs("07/22/2021", 'MM/DD/YYYY').tz("America/New_York")". The result is a timestamp that corresponds to my local time (Paris), July 22nd, 2021 0:00:00. The result I'm expecting in my local time is July 21nd, 2021 06:00:00

In other words, how can I tell my machine that the input date was created in a New York timezone ?

**Comment by addisonElliott** [comment]
@7aklhz this exact scenario is discussed in the docs here: https://day.js.org/docs/en/timezone/parsing-in-zone

Using the docs as reference, this is how you do it:

```
dayjs.tz(‘’07/22/2021, ‘MM/DD/YYYY’, ‘America/New_York’)
```

And to be clear, this is the **only** way to do it as far as I know. With your method, calling `dayjs()` will assume the machine’s local timezone. So, if I run it on my machine, the assumed timezone would be America/Chicago. Run on a different machine, the. That timezone is used. Then you call `.tz()` which converts it to that timezone.

**Comment by 7aklhz** [comment]
Thank you, I was indeed looking at the wrong document line. 

PS : for your information, I'm using dayjs in the browser.

However, I still do not get the expected result. The result I get using your answser is 07/22/2021 at 04:00. This probably happens because it (the browser ? dayjs ?) considers first "07/22/2021" to be in UTC. So, in my local time machine, that converts to 07/21/2021 at 22:00. From there, it converts it to America/New_York time and gives the result 07/22/2021 at 04:00. 

I'm able to do this correctly on my PHP backend and it seems to be because I'm setting the default timezone to UTC. However, in my browser I'm unable to to set the default timezone, even with dayjs.tz.setDefault("UTC"). 

Any ideas on how I can solve this ?

**Comment by addisonElliott** [comment]
You are not able to set the default timezone on the browser. That’s the user’s choice. You can set the default timezone in PHP because that runs on the backend rather than the browser.

Off the top of my head, I’m not sure how you can solve this. It’s not clear to me whether not specifying the time that it will default to 12:00AM like you’re expecting.

**Comment by 7aklhz** [comment]
Ok, I ended up by making an API call to my PHP server to solve this. Thank you for your time @addisonElliott
